laravel/eloquent-basic.md

```diff
@@ -0,0 +1,375 @@
+Trước khi bắt đầu, hãy chắc chắn rằng bạn đã cài đặt thông tin database trong file `.env`
+
+<a name="defining-models"></a>
+## 2. Tạo model
+
+Artisan queries:
+
+```php
+// tạo model User trong app/Http
+php artisan make:model User
+
+// tạo model đồng thời tạo file migration trong database/migrations
+php artisan make:model User --migration
+php artisan make:model User -m
+```
+<a name="eloquent-model-conventions"></a>
+## 3. Các quy tắc mặc định của Eloquent model
+<a name="table-name"></a>
+#### Tên bảng
+Mặc định, Model tên là `User` sẽ tương ứng với bảng `users`, Model `Student` sẽ tương ứng với bảng `students` và tương tự thế ...
+
+Để thay đổi tên bảng mà Model tương tác, sử dụng thuộc tính `protected $table`
+
+```php
+class Flight extends Model
+{
+   	/**
+	* The table associated with the model.
+	*
+	* @var string
+	*/
+	protected $table = 'my_flights';
+
+	public function index(){
+		//....
+	}
+	
+	//....
+}
+```
+
+<a name="primary-key"></a>
+#### Khóa chính
+
+Eloquent mặc định là mỗi bảng có một primary key tên là `id`. Để thay đổi primary key sử dụng thuộc tính `protected $primaryKey`.
+
+Ngoài ra, Eloquent mặc định rằng primary key là 1 biến int tự tăng (incremeting). Để thay đổi dùng thuộc tính `public $incrementing` và `protected $keyType`.
+
+```php
+{
+	public $incrementing = false;
+	protected $keyType = 'string';
+	protected $primaryKey = 'indentify';	
+
+	public function index(){
+		//....
+	}
+	//....
+}
+```
+
+<a name="timestamps"></a>
+#### Timestamps
+Mặc định, Eloquent cần hai cột `created_at` và `updated_at` có mặt trong các table. Nếu bạn không muốn những columns này tự động được quản lý bởi Eloquent, thiết lập thuộc tính `$timestamps` thành `false`:
+```php
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Flight extends Model
+{
+    /**
+     * Indicates if the model should be timestamped.
+     *
+     * @var bool
+     */
+    public $timestamps = false;
+}
+```
+Nếu bạn muốn thay đổi định dạng của timestamp, thay đổi thuộc tính `protected $dateFormat` trong model. Thuộc tính này xác định cách mà các thuộc tính kiểu date được lưu trong database cũng như cách format khi được serialize thành array hay JSON:
+
+```php
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Flight extends Model
+{
+    /**
+     * The storage format of the model's date columns.
+     *
+     * @var string
+     */
+    protected $dateFormat = 'U';
+}
+```
+
+<a name="retrieving-models"></a>
+## 4. Các hàm lấy dữ liệu từ DB hay sử dụng
+
+- #### `all()`: Lấy toàn bộ bản ghi
+
+```php
+$flights = App\Flight::all();
+```
+- #### `get()`: Lấy dữ liệu qua query builder
+
+```php
+$flights = App\Flight::where('active', 1)
+               ->orderBy('name', 'desc')
+               ->take(10)
+               ->get();
+```
+> Vì các Eloquent models thực chất cũng là query builders, bạn có thể sử dụng tất cả các hàm của query builder qua Eloquent model.
```

Comment:
@anhkhoahapo chỗ này làm người đọc dễ hiểu nhầm Eloquent models là query builders, em cần tìm cách diễn đạt khác ở đây

Reply:
Trong documentation của Laravel có viết là : 
> Since Eloquent models are query builders, you should review all of the methods available on the query builder. You may use any of these methods in your Eloquent queries.
https://laravel.com/docs/5.5/eloquent#retrieving-models
hay là e hiểu sai ạ ?